src/rust/bitbox02-rust/src/hww/api/ethereum/sign.rs

```diff
@@ -266,7 +278,11 @@ async fn verify_standard_transaction(
         .await?;
     }
 
-    let address = super::address::from_pubkey_hash(&recipient);
+    let address_verify = super::address::from_pubkey_hash(&recipient);
+    let address = format!("0x{}", request.address());
+    if address.to_lowercase() != address_verify.to_lowercase() {
+        return Err(Error::InvalidInput);
```

Comment:
should also fail with `InvalidInput` if `address` is not all lowercase, not all uppercase, or if mixed case, is not exactly `address_verify` (mixed case must follow the checksum algorithm).

that's why in the other comment i suggest using an enum instead which is easier to handle, as it can only contain valid values.

Reply:
Oh, I see more clearly now. I thought a string with random capitalisation was okay. Will fix that.